### PR TITLE
Adding option to keywordize documents for Firestore

### PR DESCRIPTION
**Pull request by shen-tian**
This refers to #31 and deals with the keywordizing issue as expected. 

I'd like some input on how to handle these configurable behaviour issues. In this case, we don't want to change the existing behaviour as to not break backwards compatibility. So:

1. Is a named param provided to `init` the right way to go?
2. Where should this be stored? The goto stateful place is `core/firebase-state`. Is that fine?

I'd also like to work on a PR for #28 that would require a similar flag. So some direction would be appreciated. Once I've got some input, I'll clean this up + update docs etc.

### CODE REVIEW

**Comment by deg** [comment]
I'm not sure about this one. I'm torn between not breaking existing users and not complicating the interface with additional flags.

Maybe a compromise would be:
- Combine this with a fix to #28, as you suggested.
- Add a single top-level keyword: something like `:compatibility-flags` or `:legacy-flags` that takes a set of keywords to turn on legacy behaviors.
- Maybe console.log a warning when legacy flags are used

In any event, I'd like the flag for this feature to be something more verbose than `:fs-kw`.

Basically, I want to improve the quality of the interface for new clients, while making it as easy possible for old clients to upgrade.

**Comment by shen-tian** [comment]
Think the usual thing to do is to keep adding flags for altered behaviour, and then change defaults with a new major release? In that release, we can then allow previous default behivours to be turned on as you suggest.

I don't think we should really couple this with #28.

So, would `:firestore-keywordize-keys` be a good flag name? If so, I suggest we change to that, update docs and wrap up this feature.
